src/gmail.js

```diff
@@ -3843,23 +3864,45 @@ var Gmail = function(localJQuery) {
         */
         recipients: function(options) {
             if( typeof options !== "object" ) options = {};
-            var name_selector = options.type ? "[name=" + options.type + "]" : "";
-
-            // determine an array of all emails specified for To, CC and BCC and extract addresses into an object for the callback
-            var recipients = options.flat ? [] : { to: [], cc: [], bcc: [] };
-            this.$el.find(".GS input[type=hidden]"+name_selector).each(function(idx, recipient ){
-                if(options.flat) {
-                    recipients.push(recipient.value);
+            const peopleKit = api.check.is_peoplekit_compose(this.$el);
+            
+            const type_selector = options.type ?
+                peopleKit ?
+                    "div[name=" + options.type + "] " :
+                    "[name=" + options.type + "]" : "";
```

Comment:
Right now we can provide a more specific selector for peoplekit (`div` element), but is it actually *required*?

If we don't need it, the code without that variance would be a lot simpler.

Reply:
The new code does indeed look better. Thanks!